PROTO_tests/tests/indexing_test.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+NUM_TYPES = [ak.int64, ak.uint64, ak.float64, ak.bool, ak.bigint]
+BIGINT_AND_UINT_TYPES = [ak.bigint, ak.uint64]
+
+
+def key_arrays(size):
+    ikeys = ak.arange(size)
+    ukeys = ak.arange(size, dtype=ak.uint64)
+    return ikeys, ukeys
+
+
+def value_array(num_types, size):
+    array_dict = {
+        ak.int64: (i := ak.randint(0, size, size)),
+        ak.uint64: (u := ak.cast(i, ak.uint64)),
+        ak.float64: ak.array(np.random.randn(size)),
+        ak.bool: (i % 2) == 0,
+        ak.bigint: ak.cast(u, ak.bigint),
+        ak.str_: ak.cast(i, str),
+    }
+    return array_dict[num_types]
+
+
+class TestIndexing:
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("num_types", NUM_TYPES)
+    def test_pdarray_uint_indexing(self, prob_size, num_types):
+        # for every pda in array_dict test indexing with uint array and uint scalar
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(num_types, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_strings_uint_indexing(self, prob_size):
+        # test Strings array indexing with uint array and uint scalar
+        ikeys, ukeys = key_arrays(prob_size)
+        pda = value_array(ak.str_, prob_size)
+        assert pda[np.uint(2)] == pda[2]
+        assert pda[ukeys].to_list() == pda[ikeys].to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", BIGINT_AND_UINT_TYPES)
+    def test_bool_indexing(self, prob_size, dtype):
+        pda_test = value_array(dtype, prob_size)
+        pda_uint = ak.cast(pda_test, ak.uint64)
+        pda_bool = value_array(ak.bool, prob_size)
+        assert pda_uint[pda_bool].to_list() == pda_test[pda_bool].to_list()
```

Comment:
since the left hand side is always `uint` and the right hand side is `uint` or `bool` theres a case where we are essentially comparing

```python
pda_uint[pda_bool].to_list() == pda_uint[pda_bool].to_list()
```

I think a better way of doing this (especially if we update `make_array` to pick more representative examples). Is to just get a random uint array and construct int and bigint from that
```python
def test_bool_indexing(self, prob_size):
    u = make_array(ak.uint64, prob_size)
    b = make_array(ak.bool, prob_size)
    assert u[b].to_list() == ak.cast(u, ak.int64)[b].to_list()
    assert u[b].to_list() == ak.cast(u, ak.bigint)[b].to_list()
```

Notice we no longer have a need for `BIGINT_AND_UINT_TYPES`

Reply:
Updated